cmd/lpc55-pfr/src/lib.rs

```diff
@@ -0,0 +1,320 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! ## `humility lpc55-pfr`
+//!
+//! Allows for modification of the LPC55 protected flash region (PFR) via
+//! SWD. Also includes commands for displaying the values of the protected
+//! flash region.
+//!
+//! ```code
+//!
+//! $ humility lpc55-pfr read-cfpa
+//! humility: attached via CMSIS-DAP
+//! Last scratch version: 33
+//! PONG version 33 is active
+//! Debug settings pin fd0002ff dflt ff3000cf
+//! Active RKTH revoke: 5
+//! Active image revoke: 0
+//!
+//! $ humility lpc55-pfr write-cfpa CFPA_enabled.bin
+//! humility: attached via CMSIS-DAP
+//! Wrote CFPA! Resetting chip now
+//! ```
+
+use anyhow::{Context, Result};
+use clap::Command as ClapCommand;
+use clap::{CommandFactory, Parser};
+use humility::cli::Subcommand;
+use humility::{arch::ARMRegister, core::Core};
+use humility_cmd::{Archive, Attach, Command, Validate};
+use lpc55_areas::{CFPAPage, CMPAPage};
+use std::io::Read;
+use std::path::PathBuf;
+
+#[derive(Parser, Debug)]
+#[clap(name = "subcmd")]
+enum PfrCmd {
+    /// Read the CFPA region
+    ReadCFPA,
+    /// Write the CFPA region
+    WriteCFPA {
+        #[clap(parse(from_os_str))]
+        file: PathBuf,
+    },
+    /// Read CMPA region
+    ReadCMPA,
+    /// Write CMPA region (can only be done if not locked)
+    WriteCMPA {
+        #[clap(parse(from_os_str))]
+        file: PathBuf,
+    },
+    /// Erase CMPA (can only be done if not locked)
+    EraseCMPA,
+    /// Read Keystore
+    ReadKeystore,
+}
+
+#[derive(Parser, Debug)]
+#[clap(name = "lpc55-pfr", about = env!("CARGO_PKG_DESCRIPTION"))]
+struct PfrArgs {
+    #[clap(subcommand)]
+    cmd: PfrCmd,
+}
+
+// Holds flash_config_t (see NXP manual)
+const FLASH_CONFIG_ADDR: u32 = 0x1400_0000;
```

Comment:
Probably not without jumping through some hoops. `lpc55-pac` is designed to be run on target and I don't know how it would deal with just accessing the addresses of structures.